### PR TITLE
DASBO-7421  upadte authentification to datadome api from query param to using the…

### PR DESCRIPTION
**Pull request by Simoouqas**
For this Pr we add a new authetication methode with apikey to use in the header as x-api-key,
we keep using the authetificaiton through the queryParam .
both authentication method should be available for customer but we recommend the header method .
documentation is updated .
for more details you can go here [DASBO 7421](https://datadome.atlassian.net/jira/software/projects/DASBO/boards/31?customFilter=ad0fa015-eff2-4e51-bab5-6e75b95ae608&selectedIssue=DASBO-7421)

_**IT**_ SHOULD NOT BE DEPLOYED UNTIL THE DASHBOARD TICKET IS ON PROD

### CODE REVIEW

**Comment by Simoouqas** [comment]
Please do not merge until the linked dashboard ticket is deployed

File: datadome-client-go/client.go [hunk 1]
```diff
@@ -39,8 +39,9 @@ func (c *Client) doRequest(req *http.Request, httpResponse *HttpResponse) (*Http
 	// Add apikey as a query parameter on each request for authentication
 	// Add also withoutTraffic parameter to true to have better performances
 	q := req.URL.Query()
-	q.Add("apikey", c.Token)
-	q.Add("withoutTraffic", "true")
+	//q.Add("apikey", c.Token)
+	req.Header.Set("x-api-key", "value")
+	q.Add("withoutTraffic", c.Token)
```
Comment (BastienLedon) [inline datadome-client-go/client.go:44]: That doesn't seems ok. Why "value" and why do you send the token in the query param without Traffic ?

**Review by BastienLedon — CHANGES_REQUESTED** [review]

**Comment by Lauredg** [comment]
Hello @Simoouqas ,
The integration team has just acquired responsibility for this provider, as it is a customer facing component.
Could you please update the description to give me some context and link the Jira ticket so that I can review your PR? Thanks a lot!

**Review by Lauredg — COMMENTED** [review]
LGTM! However I would like to test it before merging it. 
When the API change is in production, we will test it, merge it and then I will make the release if that's ok with you @Simoouqas .

**Review by emmanuelCarre — APPROVED** [review]
LGTM :+1: 

@Lauredg : have you test  it?

**Review by Lauredg — APPROVED** [review]

**Comment by Lauredg** [comment]
@emmanuelCarre @Simoouqas I just tested it and it works :) 
I am going to merge it as it will solve https://github.com/DataDome/terraform-provider-datadome/issues/12

**Review by BastienLedon — APPROVED** [review]

**Pull request merged**